### PR TITLE
Make pkg/helmer expose an interface

### PR DESCRIPTION
**Pull request by qbarrand**
As `pkg/helmer` does a lot of stuff, this PR:
- makes it export a `Helmer` interface and adds some unit tests before we can remove the patches it uses
- removes one of the three remaining patches, which was a no-op.

### CODE REVIEW

**Comment by qbarrand** [comment]
/cc @pmtk @ybettan @pacevedom 
PTAL

**Comment by ybettan** [comment]
Can you please squash all the commits?

File: pkg/helmer/helmer_test.go [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+		repoConfigFile := filepath.Join(tempDir, "config.yaml")
+
+		settings := cli.New()
+
+		settings.PluginsDirectory = pluginsDir
+		settings.RepositoryConfig = repoConfigFile
+		settings.RepositoryCache = filepath.Join(tempDir, "cache")
+
+		err := helmer.NewHelmer(mockCreator, settings).AddorUpdateRepo(&entry)
```
Comment (ybettan) [inline pkg/helmer/helmer_test.go:56]: Why just we don't mock the whole `Helmer` interface? why are we mocking only the inner `Creator` interface?

**Comment by ybettan** [comment]
Can you explain what are all those new yamls?

File: pkg/helmer/helmer_test.go (hunk 1 above)
Comment (qbarrand) [reply pkg/helmer/helmer_test.go:56]: We cannot mock `Helmer` here, because we are testing its implementation.

**Comment by qbarrand** [comment]
@ybettan the new YAML is a sample chart to be used in unit tests, such as `helmer_AddorUpdateRepo`.

File: pkg/helmer/helmer_test.go (hunk 1 above)
Comment (ybettan) [reply pkg/helmer/helmer_test.go:56]: Ohh Sorry, I though we are testing a higher level.

**Pull request merged**